operational/flexera/onboarding/flexera_onboarding.pt

```diff
@@ -0,0 +1,1914 @@
+  tags "provider=flexera"
+end
+
+###############################################################################
+# Pagination
+###############################################################################
+
+pagination "pagination_flexera" do
+  get_page_marker do
+    # Pagination provided by Flexera Engineering.  2024-07.  Maybe eventually a better way to implement..?
+    body_path jq(response, 'try(.nextPage | split("skipToken=") | .[1] | sub("%3D";"=";"g") | sub("%253D";"=";"g")) catch null') # The skipToken is url encoded, so we need to decode the %3D and %253D to = .  This is not ideal implementation this seems to be enough without a true urldecode function
+  end
+  set_page_marker do
+    query "skipToken"
+  end
+end
+
+###############################################################################
+# Datasources & Scripts
+###############################################################################
+
+# Get region-specific Flexera API endpoints
+datasource "ds_flexera_api_hosts" do
+  run_script $js_flexera_api_hosts, rs_optima_host
+end
+
+script "js_flexera_api_hosts", type: "javascript" do
+  parameters "rs_optima_host"
+  result "result"
+  code <<-EOS
+  host_table = {
+    "api.optima.flexeraeng.com": {
+      ui: "app.flexera.com",
+      flexera: "api.flexera.com",
+      fsm: "api.fsm.flexeraeng.com"
+    },
+    "api.optima-eu.flexeraeng.com": {
+      ui: "app.flexera.eu",
+      flexera: "api.flexera.eu",
+      fsm: "api.fsm-eu.flexeraeng.com"
+    },
+    "api.optima-apac.flexeraeng.com": {
+      ui: "app.flexera.au",
+      flexera: "api.flexera.au",
+      fsm: "api.fsm-apac.flexeraeng.com"
+    }
+  }
+
+  result = host_table[rs_optima_host]
+EOS
+end
+
+# Get Service Account Details
+datasource "ds_service_accounts" do
+  request do
+    auth $auth_flexera
+    host val($ds_flexera_api_hosts, 'flexera')
+    path join(["/iam/v1/orgs/", rs_org_id, "/service-accounts"])
+  end
+end
+
+# Get Service Account Client Details
+datasource "ds_service_account_clients" do
+  iterate $ds_service_accounts
+  request do
+    auth $auth_flexera
+    host val($ds_flexera_api_hosts, 'flexera')
+    path join(["/iam/v1/orgs/", rs_org_id, "/service-accounts/", val(iter_item, "id"), "/clients"])
+  end
+  result do
+    encoding "json"
+    collect jq(response, ".[]") do
+      field "service_account", iter_item
+      field "clientId", jq(col_item, ".clientId")
+      field "createdAt", jq(col_item, ".createdAt")
+      field "createdBy", jq(col_item, ".createdBy")
+    end
+  end
+end
+
+datasource "ds_result_service_account" do
+  run_script $js_result_service_account, $ds_service_accounts, $ds_service_account_clients
+end
+
+script "js_result_service_account", type: "javascript" do
+  parameters "ds_service_accounts", "ds_service_account_clients"
+  result "check_results"
+  code <<-EOS
+  check_results = [];
+  // Check to see if there is at least one Service Account
+  if (ds_service_accounts.length > 0) {
```

Comment:
What about the scenario where there are no Service Accounts? Should that condition also have an output to `check_results` ? (Not sure what that would look like, just playing devil's advocate )

Reply:
Nvm, I think datasource `ds_result_fail_service_account` at 1169 covers this